mini-pools-ios/ViewControllers/PoolViewController.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  PoolViewController.swift
+//  mini-pools-ios
+//
+//  Created by Kaplún, Sebi on 12/20/17.
+//  Copyright © 2017 Kaplún, Sebi. All rights reserved.
+//
+
+import UIKit
+
+class PoolViewController: UIViewController {
+
+    @IBOutlet weak var createdLabel: UILabel!
+    @IBOutlet weak var contributionsTableView: UITableView!
+    
+    var pool : PoolModel?
+    var users : [PersonModel] = []
+    var isSyncingContribution = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        contributionsTableView.dataSource = self
+        contributionsTableView.delegate = self
+        contributionsTableView.tableFooterView = UIView()
+
+        if let pool = pool {
+            self.title = pool.name
+            createdLabel.text = "Pool created on \(pool.createdAt.shortDate())"
+            if let creator = pool.creator {
+                createdLabel.text = "\(createdLabel.text!) by \(creator.name)"
+            }
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        UsersWorker.fetchUsers{ allUsers in
+            self.users = allUsers
+        }
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        let vc = segue.destination as! AddContributionViewController
+        vc.users = users
+        vc.poolId = pool!.id
+        vc.delegate = self
+        vc.modalPresentationStyle = UIModalPresentationStyle.overCurrentContext //All objects and view are transparent
+    }
+
+}
+
+extension PoolViewController : UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if let pool = pool {
+            return  pool.contributions.count  + (isSyncingContribution ? 1 : 0)
+        }
+        return isSyncingContribution ? 1 : 0
```

Comment:
does this mimic the contribution in the list before it's synced back from the server?

Reply:
yes. It shows a loading cell